locations/spiders/nissan_europe.py

```diff
@@ -0,0 +1,65 @@
+import locale
+import re
+
+import scrapy
+
+from locations.categories import Categories
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
+
+
+class NissanEuropeSpider(scrapy.Spider):
+    name = "nissan_europe"
+    item_attributes = {"brand": "Nissan", "brand_wikidata": "Q20165", "extras": Categories.SHOP_CAR.value}
+
+    def start_requests(self):
+        for l in locale.locale_alias.keys():
+            if not re.fullmatch("[a-z]{2}_[a-z]{2}", l):
+                continue
+            l = l[:3] + l[3:].upper()
+            yield scrapy.Request(
+                url=f"https://eu.nissan-api.net/v2/publicAccessToken?locale={l}&scope=READ&proxy=*&brand=nissan&environment=prod",
+                headers={"origin": "https://www.nissan-global.com"},
+                callback=self.parse_token,
+            )
+
+    def parse_token(self, response, **kwargs):
+        token = response.json()["access_token"]
+        yield scrapy.Request(
+            url="https://eu.nissan-api.net/v2/dealers?size=-1&include=openingHours",
+            headers={"accesstoken": f"Bearer {token}"},
+            callback=self.parse_dealers,
+            dont_filter=True,
+        )
+
+    def parse_dealers(self, response, **kwargs):
+        for dealer in response.json()["dealers"]:
+            dealer["id"] = dealer.pop("dealerId")
+            item = DictParser.parse(dealer)
+            if item["state"] in ("Botswana", "Namibia"):
+                item["country"] = item.pop("state")
```

Comment:
This is my first ATP spider. Is it right to fixup data like this when I see a mistake or should it rather be mapped as is?

I'll need to find a way to do this for all the new countries too.

Reply:
Is the problem that the source data `state` field could be a South African 2nd level division OR another country name? They've mashed two data fields `country` and `state` together in one and there is no way to get a state/province in Botswana or Namibia etc?

Reply:
Yes that's right